### PR TITLE
feat(curriculum): add event hub lab

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

I have modified the original user stories a bit to improve readability.

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+                <p>Date: September 5, 2024</p>
+            </article>
+        </section>
+        <section id="past-events">
+            <h2>Past Events</h2>
+            <article>
+                <h3>Cybersecurity Summit 2024</h3>
+                <p>An event focusing on the latest trends and threats in cybersecurity. Experts discussed strategies for protecting data and ensuring privacy in an increasingly digital world.</p>
+                <p>Date: June 15, 2024</p>
+                <img src="./past-event1.jpg" alt="Image from Cybersecurity Summit 2024">
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:293]: this should be updated to point to the cdn links

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            </article>
+            <article>
+                <h3>Blockchain Expo 2024</h3>
+                <p>A comprehensive event covering the future of blockchain technology. Topics included decentralized finance (DeFi), smart contracts, and the impact of blockchain on various industries.</p>
+                <p>Date: July 20, 2024</p>
+                <img src="./past-event2.jpg" alt="Image from Blockchain Expo 2024">
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:299]: this should be updated to point to the cdn links

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+You should have a `header` element.
+
+```js
+assert.isNotNull(document.querySelector("header"));
+```
+
+Your `header` element should come after the opening `body` tag.
+
+```js
+assert.match(code, /<body>[.\n\s]*<header>/im)
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:56]:
```suggestion
assert.equal(document.querySelector("body")?.firstElementChild?.tagName, "HEADER");
```
Here we can avoid using regex.

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+```
+
+Inside the `header` element, you should have a `h1` element that contains the text `Event Hub`, and a `nav` element.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:59]:
```suggestion
Inside the `header` element, you should have a `h1` element that contains the text `Event Hub`.
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+
+The second item in the unordered list should have the `href` set to `#past-events`.
+
+```js
+const anchorElement = document.querySelectorAll("header nav>ul>li>a")[1];
+const hrefAttribute = anchorElement.getAttribute("href");
+assert.strictEqual(hrefAttribute, "#past-events");
+```
+
+You should have a `main` element after the `header` element closing.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:118]:
```suggestion
You should have a `main` element after the `header` element closing tag.
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+```js
+const sectionElements = document.querySelectorAll('body>header+main>section');
+assert.strictEqual(sectionElements.length, 2);
+```
+
+Your first `section` element should have an `id` attribute with the value `upcoming-events`.
+
+```js
+const firstSection = document.querySelectorAll('body>header+main>section')[0];
+const idAttribute = firstSection.getAttribute("id");
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:136]:
```suggestion
const idAttribute = firstSection?.getAttribute("id");
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+assert.strictEqual(idAttribute, "upcoming-events");
+```
+
+Your second `section` element should have an `id` attribute with the value `past-events`.
+
+```js
+const secondSection = document.querySelectorAll('body>header+main>section')[1];
+const idAttribute = secondSection.getAttribute("id");
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:144]:
```suggestion
const idAttribute = secondSection?.getAttribute("id");
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+assert.strictEqual(idAttribute, "past-events");
+```
+
+Inside the `#upcoming-events` section, you should have an `h2` element with the text `Upcoming Events`.
+
+```js
+const h2Element = document.querySelector('#upcoming-events h2');
+assert.strictEqual(h2Element.innerText, "Upcoming Events");
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:152]:
```suggestion
assert.strictEqual(h2Element?.innerText, "Upcoming Events");
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+
+```js
+assert.match(code, /<body>[.\n\s]*<header>/im)
+```
+
+Inside the `header` element, you should have a `h1` element that contains the text `Event Hub`, and a `nav` element.
+
+```js
+const h1Element = document.querySelector('header h1');
+assert.strictEqual(h1Element.innerText, "Event Hub");
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:63]:
```suggestion
assert.strictEqual(h1Element?.innerText, "Event Hub");
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+```js
+const firstLink = document.querySelectorAll('header nav>ul>li>a')[0];
+assert.strictEqual(firstLink.innerText, "Upcoming Events");
+```
+
+The first item in the unordered list should have the `href` set to `#upcoming-events`.
+
+```js
+const anchorElement = document.querySelectorAll("header nav>ul>li>a")[0];
+const hrefAttribute = anchorElement.getAttribute("href");
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:106]:
```suggestion
const hrefAttribute = anchorElement?.getAttribute("href");
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+assert.strictEqual(hrefAttribute, "#upcoming-events");
+```
+
+The second item in the unordered list should have the `href` set to `#past-events`.
+
+```js
+const anchorElement = document.querySelectorAll("header nav>ul>li>a")[1];
+const hrefAttribute = anchorElement.getAttribute("href");
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:114]:
```suggestion
const hrefAttribute = anchorElement?.getAttribute("href");
```

File: curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+```js
+const pElement = document.querySelectorAll('#upcoming-events article h3+p').length;
+assert.strictEqual(pElement, 2);
+```
+
+Inside the `#past-events` section, you should have an `h2` element with the text `Past Events`.
+
+```js
+const h2Element = document.querySelector('#past-events h2');
+assert.strictEqual(h2Element.innerText, "Past Events");
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:180]:
```suggestion
assert.strictEqual(h2Element?.innerText, "Past Events");
```

**Review by Dario-DC — COMMENTED** [review]
It's looking good :+1: I left a few comments.

**Review by Dario-DC — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**